src/Time/Rational.hs

```diff
@@ -0,0 +1,32 @@
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE ExplicitNamespaces   #-}
+{-# LANGUAGE KindSignatures       #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Time.Rational
+       ( Rat
+       , type (%)
+       , DivRat
+       , Gcd
+       , Normalize
+       ) where
+
+import GHC.TypeNats
+
+-- | Data structure represents the rational number.
+data Rat = MkRat Nat Nat
+
+type (%) = 'MkRat
+infixl 7 %
+
+type family DivRat (a :: Rat) (b :: Rat) :: Rat where
```

Comment:
Probably better to replace `a` and `b` with `m` and `n`. To avoid name collisions with definition of `DivRat`. I think it's better to use unique names in type signature. Also it makes it more consisten with naming in standard library:

* https://hackage.haskell.org/package/base-4.10.1.0/docs/GHC-TypeNats.html#t:-43-